lib/entities/user-ui-config.ts

```diff
@@ -0,0 +1,53 @@
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import { BasicMetaSysProps, DefaultElements, MakeRequest, SysLink } from '../common-types'
+import createUserUIConfigApi from '../create-user-ui-config-api'
+import enhanceWithMethods from '../enhance-with-methods'
+
+export type UserUIConfigProps = {
+  /**
+   * System metadata
+   */
+  sys: UserUIConfigSysProps
+
+  assetListViews: ViewFolder[]
+  entryListViews: ViewFolder[]
+}
+
+export interface UserUIConfigSysProps extends BasicMetaSysProps {
+  space: SysLink
+  environment: SysLink
+}
+
+interface ViewFolder {
+  id: string
+  title: string
+  views: View[]
+}
+
+interface View {
+  id: string
+  title: string
+  order?: {
+    fieldId: string
+    direction: 'ascending' | 'descending'
+  }
+  displayedFieldIds?: string[]
+  contentTypeId: string | null
+  searchText?: string
+  searchFilters?: [string, string, string][]
+}
```

Comment:
Can we not share someof the types with the ui-config?

Reply:
I explicitly kept the types separate in case they diverge from another.
E.g. the `View` has an additional `roles` field

Reply:
True, but we could share some parts and extend them for the use case. 
But I'm also fine with two different types 👍